### PR TITLE
Computer places ships on board

### PR DESCRIPTION
**Pull request by Himalee**

### CODE REVIEW

File: lib/board.rb [hunk 1]
```diff
@@ -1,33 +1,96 @@
...
-  def place_ship(grid)
-    mark_board(grid, "C", 2, ship_mark)
-    mark_board(grid, "C", 3, ship_mark)
-    grid
+  def place_ships_on_grid(ship_mode)
+    if ship_mode == 1
+      grid_with_one_hardcoded_ship
+    elsif ship_mode == 2
+      grid_with_one_random_ship
+    end
```
Comment (pelensky) [inline lib/board.rb]: What happens if ship_mode isn't one or two?

File: lib/board.rb [hunk 2]
```diff
@@ -1,33 +1,96 @@
...
-    grid
+  def possible_columns
+    (1..size_of_board).to_a
   end
 
-  def place_ship(grid)
-    mark_board(grid, "C", 2, ship_mark)
-    mark_board(grid, "C", 3, ship_mark)
-    grid
+  def place_ships_on_grid(ship_mode)
```
Comment (pelensky) [inline lib/board.rb]: I'm wondering if this (and associated methods) could be extracted to a computer player class?

File: lib/game.rb [hunk 3]
```diff
@@ -1,52 +1,41 @@
...
+
 class Game
 
   def initialize(display, board)
     @display = display
     @board = board
   end
 
-  def play
+  def play(ship_mode)
```
Comment (pelensky) [inline lib/game.rb:16]: 👏

File: lib/board.rb [hunk 4]
```diff
@@ -1,33 +1,96 @@
...
-    size = Math.sqrt(dimension)
+  attr_reader :dimension, :grid
+
+  def initialize(dimension)
+    @dimension = dimension
+    @grid = create_grid
+  end
+
+  def create_grid
+    possible_rows.inject({}) { |result, element| result[element] = possible_columns; result }
```
Comment (sarahabimay) [inline lib/board.rb]: If you need the semi colon then this shouldn't be on one line any more.  Swap to a do/end version.

**Pull request merged**